python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py

```diff
@@ -514,10 +514,13 @@ async def _probe_queue_lens(
                     "Failed to fetch queue length for "
                     f"{replica.replica_id}: '{t.exception()}'"
                 )
-                # If we get a RayActorError, it means the replica actor has died. This
+                # If we get an ActorDiedError, the replica actor has died. This
                 # is not recoverable (the controller will start a new replica in its
                 # place), so we should no longer consider it for requests.
-                if isinstance(t.exception(), RayActorError):
+                # We do not catch RayActorError here because that error can be
+                # raised even when a replica is temporarily unavailable.
+                # See https://github.com/ray-project/ray/issues/44185 for details.
+                if isinstance(t.exception(), ActorDiedError):
```

Comment:
let's also add an `else` block for the temporarily unavailable exception type with a more informative error message here!

Reply:
We already log the error ([code](https://github.com/ray-project/ray/blob/f80d4d330ba9080c270b7cab9d2d972669d86d66/python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py#L527))

Reply:
I added a more informative error message.